### DISCUSSION TITLE
Tapping on android taps (at least) twice

### DISCUSSION THREAD

**Discussion by Morilli** [ppy/osu#15925] [Q&A] [answered]
Was just trying out the new release on android today for fun, and noticed every time I tap any circle it automatically "misses" if the next hitcircle is below the one I just tapped (seemingly tapping twice or something).

Don't have a keyboard or mouse or whatever you could possibly connect to a mobile device so I cannot test with either of that. Must be a newish issue as I've played osu!lazer on android before without this issue.

A beatmap that should make this easy to reproduce: https://osu.ppy.sh/beatmapsets/102748#osu/271261

version used: 2021.1202.0

**Comment by peppy** [comment, marked as answer]
What device are you using? Also please provide logs if you can.

**Reply by Morilli** [reply]
I'm using a `Xiaomi Mi 10 Lite 5G` running Android 11. Logs didn't show anything unusual for me, but here they are:
[logs.zip](https://github.com/ppy/osu/files/7653799/logs.zip)

**Reply by peppy** [reply]
Would also be handy to have a video showing this (screen capture is also fine), since it seems to just be you reproducing the issue. First thoughts is that your device may be somehow reporting both mouse and touch input at once.

**Reply by Morilli** [reply]
Made a video and then realized I should probably enable the key overlay as well, and immediately noticed what's going on: Every single tap taps both B1 and B2 at the same time.

Here's a video showing both the key overlay and a debug overlay from my phone to see actual screen taps:

https://user-images.githubusercontent.com/35152647/145554041-9778084a-844c-47c4-b554-fb723d4d0154.mp4

**Reply by peppy** [reply]
Have you checked your bindings?

**Reply by Morilli** [reply]
I did just now, and even though I've never changed them they do show up as changed on every startup.

Manually unsetting every of the 4(?) bindings of the right button seems to have resolved the issue.

**Reply by peppy** [reply]
On every startup? As in they are getting reset to a bad state?

**Reply by Morilli** [reply]
They are not getting set to a bad state again from what it seems, just show the little "changed" mark at the left and pressing the "reset key bindings" button did not work initially (possibly because there were 4 out of 2 possible key bindings set).

**Reply by peppy** [reply]
Please take a video showing the full thing. Need to understand whether this is your client in a bad state, some incorrect defaults, or left over settings from a previous version you had installed.

**Reply by Morilli** [reply]
The 4 setting part is mostly likely leftover settings from a previous install, as I've updated my existing install multiple times. I would try to reset my settings if I knew in which file they are saved.

Here's a video:

https://user-images.githubusercontent.com/35152647/145561762-d9228582-39b3-4473-ae23-e1e44229841a.mp4

**Reply by peppy** [reply]
Could you please attach your `client.realm` file? It should not contain any confidential information.

**Reply by Morilli** [reply]
Here: [client.realm.zip](https://github.com/ppy/osu/files/7700843/client.realm.zip)

**Reply by peppy** [reply]
Thanks. For now if you'd like to fix the issue you can safely delete `client.realm` - it will reset your key bindings and ruleset-specific settings (there aren't too many of these right now so it should be fine).

**Reply by peppy** [reply]
For what it's worth, I took a look into this from our end and there's not really much we can do to better handle this case. Probably need some kind of warning in the key binding settings when a user binds a combination that may be detrimental to gameplay.
